orbit_predictor/predictors/pass_iterators.py

```diff
@@ -201,84 +201,68 @@ def _find_los(self, tca):
 class SmartLocationPredictor(BaseLocationPredictor):
 
     def iter_passes(self):
-        current_date = self.start_date
-        while True:
-            valid, aos, tca, los, max_elevation = self._find_next_pass(current_date)
-            current_date = los
+        # Explore all values of t every 3 minutes
+        t_values = list(range(0, int((self.limit_date - self.start_date).total_seconds()), 180))
+        elev_values = []
+        for delta_seconds in t_values:
+            elev_values.append(self._elevation(delta_seconds))
```

Comment:
The same as before, and I would use list comprehension with a generator instead of a plain for loop, which is known to be faster.
 `elev_values = array.array('d', (self._elevation(delta_seconds) for delta_seconds in t_values))`

Reply:
Done!